src/tmx/TmxUtils/src/SNMPClient.cpp

```diff
@@ -0,0 +1,306 @@
+
+        // Check response
+        if (status == STAT_SUCCESS && response->errstat == SNMP_ERR_NOERROR)
+        {
+
+            PLOG(logINFO) << "STAT_SUCCESS, received a response";
+
+            if (request_type == request_type::GET)
+            {
+                for (auto vars = response->variables; vars; vars = vars->next_variable)
+                {
+                    // Get value of variable depending on ASN.1 type
+                    // Variable could be a integer, string, bitstring, ojbid, counter : defined here https://github.com/net-snmp/net-snmp/blob/master/include/net-snmp/types.h
+                    // get Integer value
+                    if (vars->type == ASN_INTEGER && vars->val.integer)
+                    {
+                        val.type = snmp_response_obj::response_type::INTEGER;
+                        val.val_int = *vars->val.integer;
+                        PLOG(logDEBUG1) << "Integer value in object: " << val.val_int;
+                    }
+                    else if (vars->type == ASN_OCTET_STR && vars->val.string)
+                    {
+                        size_t str_len = vars->val_len;
+                        for (size_t i = 0; i < str_len; ++i)
+                        {
+                            val.val_string.push_back(vars->val.string[i]);
+                        }
+                        val.type = snmp_response_obj::response_type::STRING;
+                    }
+                    else
+                    {
+                        snmp_free_pdu(response);
+                        PLOG(logERROR) << "Received a message type which isn't an integer or string";
+                        return false;
+                    }
+                }
+            }
+            else if (request_type == request_type::SET)
+            {
+
+                if (val.type == snmp_response_obj::response_type::INTEGER)
+                {
+                    PLOG(logDEBUG1) << "Success in SET for OID: " << input_oid << " Value: " << val.val_int;
+                }
+                else if (val.type == snmp_response_obj::response_type::STRING)
+                {
+                    PLOG(logDEBUG1) << "Success in SET for OID: " << input_oid << " Value: ";
+                    for (auto data : val.val_string)
+                    {
+                        PLOG(logDEBUG1) << data;
+                    }
+                }
+            }
+        }
+        else
+        {
+            log_error(status, request_type, response);
+            return false;
+        }
+
+        if (response)
+        {
+            snmp_free_pdu(response);
+            OID_len = MAX_OID_LEN;
+        }
+
+        return true;
+    }
+
+    // Backup GET function for use with prexisting ERVCloudForwarding client
+    std::string snmp_client::SNMPGet(const std::string &req_oid)
+    {
+        snmp_pdu *response;
+
+        std::string result = "";
+        auto pdu = snmp_pdu_create(SNMP_MSG_GET);
```

Comment:
There is a member variable with the same name and overlapping scope. We should probably reconsider naming here

Reply:
Addressed https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/564/commits/a49366ea8030afa8c668ec5879cf8e2d5cf69065